feat: add C implementation for `stats/base/dists/pareto-type1/mean`

AadishJ opened the pull request:
Resolves #3860.

## Description

> What is the purpose of this pull request?

This pull request:

- Adds a C implementation of the pareto type1 mean.
- Includes benchmarks for the C implementation.
- Provides examples demonstrating the usage of the pareto type1 mean in C.


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3860

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Commit:
feat(pareto-type1-mean): adding all the required functionality for pareto type1 mean

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
fix: remove cppcheck suppress

Planeshifter committed:
fix: correct C implementation to return +Infinity when alpha <= 1, update benchmarks

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/README.md:
```diff
@@ -151,6 +151,101 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_pareto_type1_mean( alpha, beta )
+
+Evaluates the mean for a Pareto Type I distribution.
```
```suggestion
Returns the [expected value][expected-value] of a [Pareto (Type I)][pareto-distribution] distribution with parameters `alpha` (shape parameter) and `beta` (scale parameter).
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/README.md:
```diff
@@ -151,6 +151,101 @@ for ( i = 0; i < 10; i++ ) {
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        alpha = random_uniform( 1.5, 5.0 ); // alpha must be > 1 for mean to be defined
```
```suggestion
        alpha = random_uniform( 1.5, 5.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/src/main.c:
```diff
@@ -0,0 +1,41 @@
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/stats/base/dists/pareto-type1/mean.h"
```
```suggestion
#include "stdlib/stats/base/dists/pareto-type1/mean.h"
#include "stdlib/math/base/assert/is_nan.h"
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/include/stdlib/stats/base/dists/pareto-type1/mean.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Evaluates the mean for a Pareto Type I distribution with shape parameter `alpha` and scale parameter `beta`.
```
```suggestion
* Returns the mean for a Pareto Type I distribution with shape parameter `alpha` and scale parameter `beta`.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/lib/native.js:
```diff
@@ -0,0 +1,63 @@
+// MAIN //
+
+/**
+* Evaluates the mean for a Pareto Type I distribution with shape parameter `alpha` and scale parameter `beta`.
```
```suggestion
* Returns the mean for a Pareto Type I distribution with shape parameter `alpha` and scale parameter `beta`.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/lib/native.js:
```diff
@@ -0,0 +1,63 @@
+* @private
+* @param {number} alpha - shape parameter
+* @param {number} beta - scale parameter
+* @returns {number} evaluated mean
```
```suggestion
* @returns {number} mean
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/src/main.c:
```diff
@@ -0,0 +1,41 @@
+#include "stdlib/stats/base/dists/pareto-type1/mean.h"
+
+/**
+* Evaluates the mean for a Pareto Type I distribution with shape parameter `alpha` and scale parameter `beta`.
+*
+* @param alpha    shape parameter
+* @param beta     scale parameter
+* @return         evaluated mean
```
```suggestion
* Returns the mean for a Pareto Type I distribution with shape parameter `alpha` and scale parameter `beta`.
*
* @param alpha    shape parameter
* @param beta     scale parameter
* @return         mean
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/benchmark/benchmark.js:
```diff
@@ -1,7 +1,7 @@
 * @license Apache-2.0
 *
-* Copyright (c) 2018 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
```suggestion
* Copyright (c) 2018 The Stdlib Authors.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,140 @@
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		alpha[ i ] = random_uniform( 1.5, 5.0 ); // alpha must be > 1 for mean to be defined
```
```suggestion
		alpha[ i ] = random_uniform( 1.5, 5.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,65 @@
+	var i;
+
+	len = 100;
+	alpha = uniform( len, 1.5, 5.0 ); // alpha must be > 1 for mean to be defined
```
```suggestion
	alpha = uniform( len, 1.5, 5.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/pareto-type1/mean/benchmark/benchmark.js:
```diff
@@ -33,14 +32,17 @@ var mean = require( './../lib' );
 	var i;
 
+	len = 100;
+	alpha = uniform( len, 1.5, 5.0 ); // alpha must be > 1 for mean to be defined
```
```suggestion
	alpha = uniform( len, 1.5, 5.0 );
```

Planeshifter committed:
docs: update description

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Merge branch 'develop' into develop6

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
chore: minor clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (approved):
Thanks, @AadishJ!
